File: backjoon/1920.cpp
#include <bits/stdc++.h>
using namespace std;
int n, m, x;
vector<int> box1;
vector<int> box2;

bool binary(int target) {
	int left = 0;
	int right = n - 1;
	while (left <= right) {
		int mid = (left + right) / 2;
		if (box1[mid] == target) {
			return true;
		}
		else if (box1[mid] <= target) {
			left = mid + 1;
		}
		else
			right = mid - 1;
	}
	return false;
}

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> m;
		box1.push_back(m);
	}
	sort(box1.begin(), box1.end());

	cin >> m;
	for (int i = 0; i < m; i++) {
		cin >> x;
		box2.push_back(x);
	}

	for (int i = 0; i < m; i++) {
		if (binary(box2[i])) cout << 1 << '\n';
		else cout << 0 << '\n';
	}
	return 0;
}

File: backjoon/1920.py
n = int(input())
m1 = list(map(int, input().split()))
m1.sort()

m = int(input())
m2 = list(map(int, input().split()))

def binary(target):
    left = 0
    right = n-1

    while (left<=right):
        mid = (left+right)//2
        if(m1[mid] == target):
            return True
        elif m1[mid] < target:
            left = mid+1
        else:
            right = mid-1
    return False

for i in m2:
    if binary(i):
        print(1)
    else:
        print(0)
        
